lib/process.js

```
@@ -1,211 +1,208 @@
+        if (!acceptAnyways) {
+          throw new Error((stderr || stdout) + '\nTimeout exceeded')
+        }
+      }
 
-      await runCommandAndCheck(['timeout', '--signal=ABRT', '1800', 'swift', 'package', 'resolve'], packageDir, true)
+      console.log('exit code not zero when running', cmd, stdout, stderr, inspectData)
+      if (!acceptAnyways) {
+        throw new Error(stderr || stdout)
+      }
+    }
+    return { stdout, stderr, data }
+  }
 
-      var { stdout, stderr } = await runCommandAndCheck(['swift', 'package', 'describe', '--type=json'], packageDir, true)
-      try {
-        description = JSON.parse(stdout)
-      } catch (err) {
+  try {
+    await docker.pull(dockerTag)
+    const container = await docker.createContainer({ Image: dockerTag, Cmd: ['/bin/bash'], AutoRemove: true, Tty: true, platform: 'linux/amd64' })
+    await container.start()
+    _container = container
+    await runCommandAndCheck(['git', 'clone', cloneUrl, '--recurse-submodules', '--branch', branch, '--single-branch', 'package'], baseDir)
 
-      }
+    const toolsResult = await runCommandAndCheck(['swift', 'package', 'tools-version'], packageDir)
+    toolsVersion = toolsResult.stdout.trim()
 
-      var { stdout, stderr } = await runCommandAndCheck(['timeout', '--signal=ABRT', '1800', 'swift', 'package', 'show-dependencies', '--format=json'], packageDir)
-      dependencies = JSON.parse(stdout.substr(stdout.indexOf('{')))
+    await runCommandAndCheck(['timeout', '--signal=ABRT', '1800', 'swift', 'package', 'resolve'], packageDir, true)
 
-      var { stdout, stderr } = await runCommandAndCheck(['swift', 'package', 'dump-package'], packageDir)
-      swiftDump = JSON.parse(stdout)
+    const describeResult = await runCommandAndCheck(['swift', 'package', 'describe', '--type=json'], packageDir, true)
+    try {
+      description = JSON.parse(describeResult.stdout)
     } catch (err) {
-      return await onError(err)
+      // ignore JSON parse errors
```

why are we now ignoring the errors? shouldn't this also be `cleanup(err)`?